### PR TITLE
Docker gradle setup

### PR DESCRIPTION
**Pull request by hendrikebbers**
All docker tasks are defined as gradle tasks.

docker

### CODE REVIEW

**Comment by hendrikebbers** [comment]
@nathanklick @JeffreyDallas this PR creates Gradle tasks for all docker workflows. While it works fine on my machine I have some questions regarding CI/CD & DevOps:

How is the docker-compose started in GitHubActions? I only found the task that creates the image?
Do we need to change the circleCI scripts?

File: build.gradle.kts [hunk 1]
```diff
@@ -22,3 +22,83 @@ plugins {
...
+    commandLine(
+        "docker",
+        "tag",
+        "services-node:" + project.version,
+        "gcr.io/hedera-registry/services-node:" + project.version
+    )
+}
+
+tasks.register<Exec>("pushDockerImage") {
+    description = "Pushes the current tag of the docker image to the google registry"
```
Comment (nathanklick) [inline build.gradle.kts:95]: We should be pushing this image. It is deprecated and will eventually be removed from the registry. 

This needs to be removed as we do not want anyone pushing by accident.

**Review by nathanklick — CHANGES_REQUESTED** [review]
This PR breaks the CI pipeline.

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+This folder contains file that are used to start the services as docker containers.
+All docker workflows are defined and gradle tasks. No skript of file in this folder should be called 'by hand'.
```
Comment (mhess-swl) [inline docker/README.md:2]:
```suggestion
All docker workflows are defined with a corresponding gradle task. No script or file in this folder should be called 'by hand'.
```

**Review by mhess-swl — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

**Review by nathanklick — CHANGES_REQUESTED** [review]
We should use a proper Gradle plugin to manage the docker image builds:

Some examples of well-established and supported plugins:
- https://github.com/bmuschko/gradle-docker-plugin
- https://github.com/GoogleContainerTools/jib/tree/master/jib-gradle-plugin

**Comment by hendrikebbers** [comment]
@nathanklick I looked at Docker plugins and I assume that using one of them will end in more changes in the repo than the current PR. I wanted to change as less in the workflow as possible. I'm with you that using a docker plugin instead of the exec plugin should be the preferred way and would propose to do it as a next step. I will create an issue for that.

**Review by nathanklick — APPROVED** [review]

**Pull request merged**